src/Nethermind/Nethermind.Synchronization/Peers/AllocationStrategies/TransitioningPeerAllocationStrategy.cs

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Generic;
+using Nethermind.Blockchain;
+using Nethermind.Consensus;
+using Nethermind.Stats;
+
+namespace Nethermind.Synchronization.Peers.AllocationStrategies;
+
+/// <inheritdoc />
+/// <summary>
+/// Uses <see cref="TotalDiffStrategy"/> before the merge and <see cref="LastBlockStrategy"/> after.
+/// </summary>
+public class TransitioningPeerAllocationStrategy : IPeerAllocationStrategy
+{
+    private IPeerAllocationStrategy _currentStrategy;
+
+    public TransitioningPeerAllocationStrategy(IPoSSwitcher poSSwitcher, IPeerAllocationStrategy innerStrategy)
+    {
+        if (poSSwitcher.TransitionFinished)
+        {
+            _currentStrategy = CreatePostMergeStrategy(innerStrategy);
+        }
+        else
+        {
+            poSSwitcher.Transitioned += (_, _) => _currentStrategy = CreatePostMergeStrategy(innerStrategy);
```

Comment:
We can save a few bytes of statically used memory by unsubscribing everything after `Transitioned` is fired, as it's expected to trigger only once. Should we?

Reply:
Can set to `null` if nothing else should be subscribed (removes all events)